scripts.py

```diff
@@ -287,8 +287,10 @@ def parse_command_line(cls, _db, cmd_args=None, stdin=sys.stdin,
             parsed.cutoff_time = cls.parse_time(parsed.cutoff_time)
         return parsed
 
-    def __init__(self, provider):
-        args = self.parse_command_line(self._db)
+    def __init__(self, provider, _db=None, cmd_args=None, **provider_arguments):
+        if _db:
+            self._session = _db
```

Comment:
Script._db is a @property that checks self._session and instantiates a new database session (storing it as ._session) if none is present. Since we're in the middle of a test we don't want to instantiate a new session, so we make sure ._session is set before we do anything.